### ISSUE TITLE
Dynamic input ports, part 2

### ISSUE THREAD

**Issue by dcwhite** [SCIInstitute/SCIRun#445]
Continued from #388 
- [x] Multiple dynamic ports works, but adds to end. Should improve to add at adjacent position. Moved to new issue: #678
- [x] Mac d-port display is way out of place. Need platform dependent placement code most likely. Moved to new issue: #679
- [x] Check Linux display. Blocked by #677.
- [x] Refactoring broke serialization completely. Need to generalize loading of connections to respect arbitrary indices of dynamic ports.
- [x] Found a bug when user deletes a connection attached to ViewScene port 0, when executed it looks for the deleted port and errors.
- [x] On Mac, when connecting to a dynamic port, the module jumps 500 pixels to the right. :dizzy_face:
